packages/@sanity/form-builder/src/components/Details.tsx

```diff
@@ -1,50 +1,71 @@
-import {ChevronDownIcon} from '@sanity/icons'
-import React from 'react'
+import {ToggleArrowRightIcon} from '@sanity/icons'
+import {Box, Flex, Text} from '@sanity/ui'
+import React, {useCallback, useState} from 'react'
+import styled from 'styled-components'
 
-import styles from './Details.css'
-
-type DetailsProps = {
-  isOpen?: boolean
+interface DetailsProps {
+  children?: React.ReactNode
+  open?: boolean
   title?: React.ReactNode
 }
 
-type DetailsState = {
-  isOpen: any
-}
+const HeaderButton = styled.button`
+  -webkit-font-smoothing: inherit;
+  appearance: none;
+  font: inherit;
+  background: none;
+  width: 100%;
+  text-align: left;
+  border: 0;
+  margin: 0;
+  padding: 0;
+  outline: none;
+`
 
-export class Details extends React.Component<DetailsProps, DetailsState> {
-  static defaultProps = {
-    title: 'Details',
-    isOpen: false,
-  }
+const Header = styled(Flex)`
+  cursor: default;
+  user-select: none;
```

Comment:
What's the reason for `user-select: none` here? 

It seems to be different from how [native `<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) elements work.